layers/+distribution/spacemacs-base/packages.el

```diff
@@ -847,7 +852,23 @@ below. Anything else exits."
       (add-to-list 'recentf-exclude
                    (expand-file-name spacemacs-cache-directory))
       (add-to-list 'recentf-exclude (expand-file-name package-user-dir))
-      (add-to-list 'recentf-exclude "COMMIT_EDITMSG\\'"))))
+      (add-to-list 'recentf-exclude "COMMIT_EDITMSG\\'")
+
+      (defsubst spacemacs//file-was-viewed-p (file)
+        "Return non-nil if FILE's buffer exists and has been displayed."
+        (interactive)
+        (let ((buf (find-buffer-visiting file)))
+          (> (buffer-local-value 'buffer-display-count buf) 0)))
+
+      (defsubst spacemacs//recentf-viewed-predicate (file)
+        "Return non-nil if FILE was previously in recentf list or has been
+displayed during this session."
+        (and (recentf-keep-default-predicate file)
```

Comment:
If the `setf` below is replaced with `add-to-list`, then we don't need to check `recent-keep-default-predicate` here.